src/subdomains/core/sell-crypto/process/services/buy-fiat.service.ts

```diff
@@ -227,6 +237,54 @@ export class BuyFiatService {
     buyFiat.sell ? await this.webhookService.cryptoFiatUpdate(buyFiat.user, extended) : undefined;
   }
 
+  async refundBuyFiat(buyFiatId: number, dto: RefundCryptoInputDto): Promise<void> {
+    const buyFiat = await this.buyFiatRepo.findOne({ where: { id: buyFiatId }, relations: { cryptoInput: true } });
+
+    await this.refundBuyFiatInternal(
+      buyFiat,
+      { refundUserId: dto.refundUser.id, refundAddress: buyFiat.chargebackAddress },
+      dto.chargebackAmount,
+    );
+  }
+
+  async refundBuyFiatInternal(
+    buyFiat: BuyFiat,
+    chargebackDto: TransactionRefundDto,
+    chargebackAmount?: number,
+  ): Promise<void> {
+    if (!chargebackDto.refundAddress && !chargebackDto.refundUserId)
+      throw new BadRequestException('You have to define a chargebackAddress');
```

Comment:
Check is already done with DTO validation?

Reply:
Not necessarily, only it the request came from transaction controller and `TransactionRefundDto`, not necessarily if it comes from buyFiat endpoint with `RefundCryptoInputDto`. The admin does not necessarily have to assign an address with the buyFiat endpoint, but must have the option to do it

Reply:
Ok